sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/profile/AzureProfile.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.profile;
+
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.util.Configuration;
+
+/**
+ * Azure profile for client.
+ */
+public class AzureProfile {
+
+    private String tenantId;
+    private String subscriptionId;
+    private AzureEnvironment environment;
+    private final Configuration configuration = Configuration.getGlobalConfiguration();
```

Comment:
Prefer this be settable, so user can choose to use or not use the configuration.

In this code they do not have the choice at all.

Reply:
I prefer to have boolean flag/method like `allowGobalConfiguration` if we want to make it settable. I was thinking about to have a builder class, just not sure if we should apply the style of  `with*()` to builder class here.

Reply:
Update?